### DISCUSSION TITLE
how do you convert string to int and vice versa?

### DISCUSSION THREAD

**Discussion by MrMidnightOfficial** [vlang/v#14883] [General]
how do you convert string to int and vice versa?

**Comment by medvednikov** [comment]
```
number := s.int()
s := number.str()
```

**Reply by MrMidnightOfficial** [reply]
> ```
> number := s.int()
> s := number.str()
> ```

TYSM you have been a great help

**Reply by chikega** [reply]
> number := s.int()
> s := number.str()

I get this error when I attempt to use `u8`: `error: unknown method or field: string.u8`
 `u16`,`u32` and`u64` all work as expected.

```
age_s := os.input("How old are you? ")
age_i := age.u8()
```

**Reply by Delta456** [reply]
> > number := s.int()
> > s := number.str()
> 
> I get this error when I attempt to use `u8`: `error: unknown method or field: string.u8`
>  `u16`,`u32` and`u64` all work as expected.
> 
> ```
> age_s := os.input("How old are you? ")
> age_i := age.u8()
> ```

I think you need to do `age.byte()` instead if it doesn't work then it must be implemented.

**Reply by chikega** [reply]
It appears the same error is generated with `age.byte()`:  `error: unknown method or field: 'string.byte'.`

**Comment by JalonSolov** [comment]
It appears to be working fine with current V (and corrected program):
```v
import os

age_s := os.input("How old are you? ")
age_i := age_s.u8()
println(age_i)
```
```
$ v run foo.v
How old are you? 99
99
$
```

**Reply by JalonSolov** [reply]
If it doesn't work for you, make sure you have at least `V 0.3.1 8666ef4`.  If not, do `v up` to get the latest.

**Reply by chikega** [reply]
Thank you Jalon, 
That did the trick. Homebrew installs the latest 'stable version' (0.2.4 b72a2de). So I downloaded the beta version from the main page download link for mac - V 0.3.1 267e26b. After futzing about with permissions on mac os, I finally got it to work. Although` age_s.byte() `does not work.
Yes, and my code should have shown the consumption of the variable as well.  Here's the entire simple I/O program that I was testing out in V: 
```
import os 

fn main() {
	fname := os.input("What is your name? ")
	println("It's nice to meet you $fname!")
	age_s := os.input("How old are you? ") 	 
	mut age_i := age_s.u8()   		
	print(age_s) // verfiying that no newline is captured in the stream - so, no need for trim function 
	gpa_s := os.input("\nWhat is your GPA? ")
	mut gpa_f := gpa_s.f32()  
	println("So $fname, you are $age_i years old and your GPA is $gpa_f .. stellar!")
}
```

**Reply by JalonSolov** [reply]
V doesn't have any official packages.  Whoever created that homebrew package obviously hasn't maintained it since.  :-\

That's why we have this:

https://github.com/vlang/v#installing-v---from-source-preferred-method

**Reply by chikega** [reply]
Thanks again Jalon, that really helps. Easier to work on my PC than mac .. had to allow permissions in the 'Security & Privacy' pane to run `v`, `v up`, `tcc` .. but finally got V up to `V 0.3.1 8666ef4` on my mac. Cheers! 🤓
